keanu-project/src/main/java/io/improbable/keanu/templating/SequenceItem.java

```diff
@@ -54,7 +54,11 @@ public SequenceItem() {
             throw new IllegalArgumentException("Key " + label + " already exists");
         }
         contents.put(label, v);
-        v.setLabel(label);
+        if (v instanceof ProxyVertex) {
+            ((ProxyVertex) v).addScopeToLabel(getUniqueName());
```

Comment:
This was because if we call `setLabel` on a ProxyVertex to add the scope (This is the only time it's acceptable to call `setLabel on a proxy vertex`) then it will throw an exception. The `addScopeToLabel` is a safe way of doing the same thing.